operator/cnp_event.go

```diff
@@ -48,7 +53,20 @@ func enableCNPWatcher() error {
 			AddFunc: func(obj interface{}) {
 				metrics.EventTSK8s.SetToCurrentTime()
 				if cnp := k8s.CopyObjToV2CNP(obj); cnp != nil {
-					groups.AddDerivativeCNPIfNeeded(cnp.CiliumNetworkPolicy)
+					if !cnp.RequiresDerivative() {
+						log.WithFields(logrus.Fields{
+							logfields.CiliumNetworkPolicyName: cnp.ObjectMeta.Name,
+							logfields.K8sNamespace:            cnp.ObjectMeta.Namespace,
+						}).Debug("CNP does not have derivative policies, skipped")
+						return
+					}
+
+					controllerManager.UpdateController(fmt.Sprintf("add-derivative-cnp-%s", cnp.ObjectMeta.Name),
+						controller.ControllerParams{
+							DoFunc: func(ctx context.Context) error {
+								return addDerivativeCNP(cnp.CiliumNetworkPolicy)
```

Comment:
Why is this inside a controller and not a serializer queue? Having it as controllers could cause issues concurrently for a CNP with the same name. A serializer queue will make sure an event is processed before going to the next one.

Reply:
Good point!